tests/test_stream.py

```diff
@@ -0,0 +1,420 @@
+# Copyright 2013-2018 Donald Stufft and individual contributors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from __future__ import absolute_import, division, print_function
+
+import binascii
+
+import pytest
+
+from nacl.bindings.crypto_stream import (
+    crypto_stream_chacha20,
+    crypto_stream_chacha20_ietf,
+    crypto_stream_chacha20_ietf_keygen,
+    crypto_stream_chacha20_ietf_xor,
+    crypto_stream_chacha20_ietf_xor_ic,
+    crypto_stream_chacha20_keygen,
+    crypto_stream_chacha20_xor,
+    crypto_stream_chacha20_xor_ic,
+    crypto_stream_xchacha20,
+    crypto_stream_xchacha20_KEYBYTES,
+    crypto_stream_xchacha20_NONCEBYTES,
+    crypto_stream_xchacha20_keygen,
+    crypto_stream_xchacha20_xor,
+    crypto_stream_xchacha20_xor_ic,
+    has_crypto_stream_xchacha20,
+)
+from nacl.exceptions import UnavailableError
+from nacl.utils import random as randombytes
+
+
+def sodium_is_zero(data):
```

Comment:
Please add a comment explaining that this is a port of the libsodium function of the same name so future reviewers won't wonder why this exists